File: 145-二叉树的后序遍历.cpp
class Solution {
public:
    vector<int> postorderTraversal(TreeNode* root) {
        stack<TreeNode*> mystack;
        vector<int> result;
        mystack.push(root);

        while(!mystack.empty()){
            root = mystack.top();
            mystack.pop();
            if(root){
                result.push_back(root->val);
                mystack.push(root->left);
                mystack.push(root->right);
            }
        }

        reverse(result.begin(), result.end());

        return result;
    }
};

File: 145-二叉树的后序遍历.py
class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:

        result = []

        def postorder(root):
            if root:
                postorder(root.left)
                postorder(root.right)
                result.append(root.val)
        
        postorder(root)

        return result
